phac-nml/irida
# Update REST API status code returned when adding files to the wrong type of run

## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Added try/catch blocks when uploading sequence files to return a bad request status code (400) if an IllegalArgumentException is thrown, or return an Internal Server Error status code (500) for any other exceptions thrown.

To test this:

1) Get the latest version of the IRIDA Uploader
2) Upload a paired end sequence using the uploader
3) Go into the folder which you uploaded the paired end sequence from, open up the irida_uploader_status.info file, and get the run_id
4) Modify the api_handler.py file under the core/ directory as follows:
- Replace this line: `run_id = api_instance.create_seq_run(sequencing_run.metadata)` with `run_id=run_id_retrieved_from_file_above`
5) Now upload a single-end sequence

You should see something similar to:

2019-04-10 15:21:28 ERROR    Error while uploading [08-5578]: [Bad Request]
2019-04-10 15:21:28 ERROR    Failed to upload SequencingRun, Could not connect to IRIDA
2019-04-10 15:21:28 ERROR    Lost connection to Irida
2019-04-10 15:21:28 ERROR    Errors: ('Error 400: Bad Request\n',)

Note: An issue has been made on the Irida Uploader Github page to provide a clear message upon a bad request rather than `Could not connect to IRIDA` and `Lost connection to IRIDA`


## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

Fixes #178 

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~~* [ ] Tests added (or description of how to test) for any new features.~~
~~* [ ] User documentation updated for UI or technical changes.~~
